### ISSUE TITLE
Need a way to retrieve console output automatically

### ISSUE THREAD

**Issue by JKereliuk** [w3c/web-platform-tests#8654]
Maybe we could wrap console.log with a hook?

for example: [https://github.com/w3c/web-platform-tests/blob/master/console/console-count-logging-manual.html](https://github.com/w3c/web-platform-tests/blob/master/console/console-count-logging-manual.html)

@domfarolino WYT?

**Comment by domfarolino** [comment]
Hey, sorry could you give me some more context here? Not sure I quite understand!

**Comment by domfarolino** [comment]
I definitely see the motivation of wanting console output, but I don't see a feasible way of capturing its output at all. Of course we could wrap `console` methods in hooks, and store their arguments somewhere, but that doesn't get us close to actually capturing the *output*, thoughts?

**Comment by OrKoN** [comment]
I believe this is possible with WebDriver BiDi: https://github.com/web-platform-tests/wpt/pull/47478 cc @sadym-chromium

**Comment by foolip** [comment]
This should indeed be possible to test with https://w3c.github.io/webdriver-bidi/#event-log-entryAdded which is already wrapped in testdriver.js as documented in https://web-platform-tests.org/writing-tests/testdriver.html#entry-added.

However, it seems like no console tests have been converted to use this, so converting some manual tests in https://github.com/web-platform-tests/wpt/tree/master/console to show that it really works would be a good idea before closing this.

**Comment by OrKoN** [comment]
FYI @foolip there is an example in infrastructure tests https://github.com/web-platform-tests/wpt/blob/master/infrastructure/testdriver/bidi/subscription.html

**Comment by foolip** [comment]
@OrKoN yes I saw those, but thought automating some of the console tests would be the ultimate proof this is now a solved problem.

**Comment by Minju-kim-igalia** [comment]
I’m working on converting some of the existing manual console tests to automated tests using WebDriver BiDi `log.entryAdded`via `testdriver.js?feature=bidi`.

Chromium CL (in progress): https://chromium-review.googlesource.com/c/chromium/src/+/7657323

**Comment by Minju-kim-igalia** [comment]
Update: This is now testable via WebDriver BiDi `log.entryAdded`, which is wrapped in `testdriver.js?feature=bidi`, so WPT can observe console output automatically.
I have converted serveral manual WPT console tests to BiDi-based automation in Chromium (https://crrev.com/c/7657323).
Given the above, I think the main ask of this issue is addressed and it can likely be closed as resolved.

Side note: While converting tests, I noticed a remaining discrepancy for `console.dirxml("%s", Symbol("description"))`: BiDi `log.entryAdded.text` can differ from other console methods (e.g. showing "symbol" rather than "Symbol(description)").
Based on discussion in chromium-bidi side ([PR #4123](https://github.com/GoogleChromeLabs/chromium-bidi/pull/4123)), this may be better handled as a spec level discussion:  https://github.com/w3c/webdriver-bidi/issues/89

**Comment by domfarolino** [comment]
Thanks so much for working on this @Minju-kim-igalia! This is great, I remember seeing this issue open when I was in college, wondering what a future solution to all my "manual" tests might be, and I'm glad we've wound up in a much better place!

> Given the above, I think the main ask of this issue is addressed and it can likely be closed as resolved.

I agree, I'll close this.
